### PR TITLE
Juju to now use k8s client-go config loading.

### PR DESCRIPTION
**Pull request by tlm**
Up until this stage we have been doing our own form of config file
loading for Kubernetes configuration libaries. Because of this we have
not been handling multiple files separated by semicolon's in KUBECONFIG
env variable correctly.

This commit addresses the above by changing Juju to use the default
tools built into the k8s.io/client-go package for loading configuration
files. This makes sure that Juju observes the correct standards set out
by Kubernetes and their tooling.

## Checklist

 - [x] Requires a [pylibjuju](https://github.com/juju/python-libjuju) change
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed
 - [x] Comments answer the question of why design decisions were made

## QA steps

Add multiple file definitions to a KUBECONFIG env variable and run juju add-k8s

```sh
export KUBECONFIG="/home/tlm/.kube/config.yaml;/home/tlm/.kube/config1.yaml"
juju add-k8s <some-context>
# Before this change this would have failed. Now it should succeed without error.
```

### CODE REVIEW

**Comment by tlm** [comment]
[Fix juju so that it can understand multiple kubeconfig files.](https://trello.com/c/yHf1r4xk/1347-fix-juju-so-that-it-can-understand-multiple-kubeconfig-files)

File: caas/kubernetes/clientconfig/k8s.go [hunk 1]
```diff
@@ -40,22 +39,15 @@ func NewK8sClientConfig(
 	contextName, clusterName string,
 	credentialResolver K8sCredentialResolver,
 ) (*ClientConfig, error) {
-	if reader == nil {
```
Comment (wallyworld) [inline caas/kubernetes/clientconfig/k8s.go:43]: reader now appears unused in the new code
i suspect there will be failing tests now also

Comment (tlm) [reply caas/kubernetes/clientconfig/k8s.go:43]: Done & Done

**Review by hpidcock — APPROVED** [review]

**Comment by tlm** [comment]
$$merge$$

**Pull request merged**